src/Shared/Contracts/IManagerProvider.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+
+namespace Microsoft.CST.OpenSource.Contracts;
+
+using PackageManagers;
+using PackageUrl;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+/// <summary>
+/// The interface to implement project manager specific methods.
+/// Useful for mocking when doing unit tests.
+/// </summary>
+public interface IManagerProvider<T>
+{
+    /// <summary>
+    /// The <see cref="IHttpClientFactory"/> for this <see cref="IManagerProvider{T}"/> to make <see cref="HttpClient"/>s with.
+    /// </summary>
+    public IHttpClientFactory HttpClientFactory { get; }
```

Comment:
Since many of the managers may use HttpClientFactory behind the scenes I think it makes sense to have an `IHttpClientBasedManagerProvider` or `BaseHttpClientBasedManagerProvider` that does have this field to make it easier to develop those. But agree the base interface does not need it.

Reply:
Removed HttpClientFactory from the IManagerProvider.